src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/InstantiateCommand.cs

```diff
@@ -172,6 +173,26 @@ async Task<string> GetTemplatePackagesList(TemplateGroup templateGroup)
             }
         }
 
+        // Reparses the CLI parameters in the context of the provided command (usually TemplateCommand).
+        // If the first token after reparsing starts with a dash, we reparse again and inject a name sentinel to allow it to be handled properly (using empty name logic).
+        internal static ParseResult Reparse(CliCommand command, IReadOnlyList<string> args, CliConfiguration? configuration = null)
+        {
+            var argsList = args.ToList();
+            ParseResult parseResult = command.Parse(args, configuration);
+
+#pragma warning disable CA1826
+            var firstTemplateCommandToken = parseResult.CommandResult.Tokens.FirstOrDefault()?.Value ?? string.Empty;
+#pragma warning restore CA1826
```

Comment:
Not sure why I was getting a CA1826 here as the logic requires `FirstOrDefault` since I'm not sure if there even is a token available.